draft-ietf-quic-recovery.md

```diff
@@ -422,6 +422,60 @@ is available.
 (TODO: Work this section some more. Add text on client vs. server, and on
 stateless retry.)
 
+## Generating Acknowledgements
+
+QUIC SHOULD delay sending acknowledgements in response to packets,
+but MUST NOT excessively delay acknowledgements of packets containing
+non-ack frames.  Specifically, implementaions MUST attempt to
+enforce a maximum ack delay to avoid causing the peer spurious
+timeouts.  The default maximum ack delay in QUIC is 25ms.
+
+An acknowledgement MAY be sent for every second full-sized segment,
+as TCP does {{!RFC5681}}, or may be sent less frequently, as long as
+the delay does not exceed the maximum ack delay.  QUIC recovery algorithms
+do not assume the peer generates an acknowledgement immediately when
+receiving a second full-sized segment.
+
+Out-of-order packets SHOULD be acknowledged more quickly, in order
+to accelerate loss recovery.  Specifically, the receiver SHOULD send an
+immediate ACK when it receieves a packet with a packet number more than
+1 larger than the previous largest received packet number. The receiver
+SHOULD also send an immediate ACK when it receives a packet that fills
+in a gap in the packet numbers.  If no packet larger than the packet that
```

Comment:
might i suggest a different wording.

The receiver SHOULD also send an immediate ACK when it receives a packet that is not the next expected packet number.

This is how we implement this.

Reply:
i.e. i think this covers the case of a gap being filled.

Reply:
Agreed that's a good algorithm.  Updated and I tried to reword the bulk receiving text to be clearer.  Suggestions welcome.

Reply:
the changes look great